src/fidesops/schemas/masking/masking_secrets.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC
+from dataclasses import dataclass
+from enum import Enum
+from typing import TypeVar, Generic, Callable, Optional
+
+from fidesops.schemas.base_class import BaseSchema
+
+
+T = TypeVar("T")
+
+
+class SecretType(Enum):
+    """Enum that holds all possible types of secrets across all masking strategies"""
+
+    key = "key"
+    salt = "salt"
+    # the below types are used by the AES algorithm, when it calls HMAC to generate the nonce
+    key_hmac = "key_hmac"
+    salt_hmac = "salt_hmac"
+
+
+@dataclass(unsafe_hash=True)
+class MaskingSecretMeta(Generic[T]):
+    """Holds metadata describing one secret"""
+
+    masking_strategy: str
+    secret_type: SecretType
+    generate_secret: Callable[[int], T]
```

Comment:
Would be good for the naming here to indicate this is a function too, since when you see `generate_secret=...` inside kwargs that isn't immediately obvious